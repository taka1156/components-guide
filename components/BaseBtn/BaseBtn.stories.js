import BaseBtn from './BaseBtn.vue';

export default {
  title: 'Atoms/BaseBtn',
  component: BaseBtn,
  argTypes: {
    btnClick: { action: 'clicked' },
    size: {
      control: {
        type: 'inline-radio',
        options: ['sm', 'md', 'lg']
      }
    },
    btnType: {
      control: {
        type: 'inline-radio',
        options: ['fill', 'outlined']
      }
    }
  }
};

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { BaseBtn },
  template: '<base-btn v-bind="$props" @btn-click="btnClick">Button</base-btn>'
});

export const Default = Template.bind({});
Default.args = { size: 'sm', btnType: 'outlined' };
