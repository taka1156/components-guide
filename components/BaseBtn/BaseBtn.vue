<template>
  <div>
    <button
      type="button"
      class="base-btn--extend"
      :class="`base-btn--${size} base-btn--${btnType}`"
      @click="btnClick"
    >
      <!-- @slot ボタンラベル -->
      <slot />
    </button>
  </div>
</template>

<script>
export default {
  name: 'BaseBtn',
  props: {
    size: {
      type: String,
      default: 'md',
      validator: value => {
        return ['sm', 'md', 'lg'].indexOf(value) !== -1;
      }
    },
    btnType: {
      type: String,
      default: 'fill',
      validator: value => {
        return ['fill', 'outlined'].indexOf(value) !== -1;
      }
    }
  },
  methods: {
    btnClick() {
      /**
       * ボタンクリックイベント
       * @event btnClick
       * @type {event}
       */
      this.$emit('btn-click');
    }
  }
};
</script>

<style scoped>
/* css reset */
button {
  padding: 10px;
  margin: 0;
  line-height: 50%;
  cursor: pointer;
  border-radius: 5px;
}

/* css reset */

.base-btn--fill {
  color: white;
  background-color: green;
  border: none;
}

.base-btn--fill:hover {
  opacity: 0.85;
}

.base-btn--outlined {
  color: green;
  background-color: white;
  border: 1px solid green;
}

.base-btn--outlined:hover {
  color: white;
  background-color: green;
  border: none;
}

.base-btn--sm {
  width: 100px;
  height: 30px;
  font-size: 15px;
}

.base-btn--md {
  width: 150px;
  height: 40px;
  font-size: 20px;
}

.base-btn--lg {
  width: 200px;
  height: 50px;
  font-size: 25px;
}
</style>
