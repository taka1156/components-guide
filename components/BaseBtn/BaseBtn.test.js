import { mount } from '@vue/test-utils';
import BaseBtn from './BaseBtn.vue';

describe('BaseBtn', () => {
  const dummyButtonText = 'ダミーボタンテキスト';
  const baseBtn = propsData =>
    mount(BaseBtn, {
      props: {
        ...propsData
      },
      slots: {
        default: dummyButtonText
      }
    });

  it('slotがDOMに反映されているか', () => {
    const btnTag = baseBtn();
    expect(btnTag.text()).toBe(dummyButtonText);
    // スナップショット
    expect(btnTag.find('button').html()).toMatchSnapshot();
  });

  it('buttonサイズ:sm', () => {
    const btnTag = baseBtn({ size: 'sm' });
    // スナップショット
    expect(btnTag.html()).toMatchSnapshot();
  });

  it('sbuttonサイズ:lg', () => {
    const btnTag = baseBtn({ size: 'lg' });
    // スナップショット
    expect(btnTag.html()).toMatchSnapshot();
  });

  it('btnClickを実行した時、`btn-click`のイベントが発生するか', () => {
    const btnTag = baseBtn({ size: 'lg' });
    btnTag.vm.btnClick();
    expect(btnTag.emitted('btn-click')).toBeTruthy();
  });
});
