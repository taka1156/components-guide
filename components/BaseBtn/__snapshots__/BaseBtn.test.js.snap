// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseBtn buttonサイズ:sm 1`] = `"<div><button type=\\"button\\" class=\\"base-btn--extend base-btn--undefined base-btn--fill\\">ダミーボタンテキスト</button></div>"`;

exports[`BaseBtn sbuttonサイズ:lg 1`] = `"<div><button type=\\"button\\" class=\\"base-btn--extend base-btn--undefined base-btn--fill\\">ダミーボタンテキスト</button></div>"`;

exports[`BaseBtn slotがDOMに反映されているか 1`] = `"<button type=\\"button\\" class=\\"base-btn--extend base-btn--md base-btn--fill\\">ダミーボタンテキスト</button>"`;
